src/transformers/modeling_utils.py

```diff
@@ -589,8 +589,9 @@ def invert_attention_mask(self, encoder_attention_mask: Tensor) -> Tensor:
 
         return encoder_extended_attention_mask
 
-    def create_extended_attention_mask_for_decoder(self, input_shape, attention_mask, device):
+    def create_extended_attention_mask_for_decoder(self, input_shape, attention_mask):
```

Comment:
Removing an argument from a public method like this is a breaking change, so we should continue to accept if with a default of `None` than raise a deprecation warning if we detect it's not `None` telling the user that argument is not used anymore and will be removed in v5 of Transformers.